tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -180,29 +216,32 @@ internal async override void InvokeTool(Window? parentWindow, int? targetProcess
                     {
                         // Note: running most msc files requires elevation.
                         finalExecutable = "mmc.exe";
-                        finalArguments = $"{Executable} {parsedArguments}";
+                        finalArguments = $"{Executable} {commandLineArgs}";
                     }
                     else if (Path.GetExtension(Executable).Equals(".ps1", StringComparison.OrdinalIgnoreCase))
                     {
                         // Note: running powershell scripts might require setting the execution policy.
                         finalExecutable = "powershell.exe";
-                        finalArguments = $"{Executable} {parsedArguments}";
+                        finalArguments = $"{Executable} {commandLineArgs}";
                     }
                     else
                     {
                         finalExecutable = Executable;
-                        finalArguments = parsedArguments;
+                        finalArguments = commandLineArgs;
                     }
 
                     var startInfo = new ProcessStartInfo()
                     {
                         FileName = finalExecutable,
                         Arguments = finalArguments,
-                        UseShellExecute = true,
+
+                        // If we want to redirect standard out, we can't use shell execute
+                        UseShellExecute = !options.RedirectStandardOut,
+                        RedirectStandardOutput = options.RedirectStandardOut,

```

Comment:
this also means you would need to handle the elevation case manually for these types of tools

Reply:
Yeah... I guess the alternative is to pipe it to temp files? RunUtility.exe > c:\temp\output.txt?

Reply:
right what is what you would have to do for an elevated tool anyway since I don't believe you can redirect standard input/output for a process that is running elevated. I was just noting we now have a tool execution path that can fail that didn't before.

Reply:
Yep. Fortunately the only feature that goes down this code path is the debugger analysis, and I'm told you don't want these things to run elevated.

Maybe in cases where we don't need to go elevated, we use the stdout pipe, those that do write out to a temp file? I'm assuming the pipe will be more reliable, so if we can use it, we should.

Something to keep in mind for the external tools story.